DQM/GEM/plugins/GEMDQMStatusDigi.cc

```diff
@@ -0,0 +1,361 @@
+    ( (TH2F *)GEMDAV2D->getTH2F() )->GetYaxis()->SetBinLabel(nNumAMC,tmpLabel);
+  
+    ( (TH2F *)Tstate2D->getTH2F() )->SetBins(15, 0, 15, nNumAMC, 0, nNumAMC);
+    ( (TH2F *)Tstate2D->getTH2F() )->GetYaxis()->SetBinLabel(nNumAMC,tmpLabel);
+  
+    ( (TH2F *)GDcount2D->getTH2F() )->SetBins(32, 0, 32, nNumAMC, 0, nNumAMC);
+    ( (TH2F *)GDcount2D->getTH2F() )->GetYaxis()->SetBinLabel(nNumAMC,tmpLabel);
+  
+    ( (TH2F *)ChamT2D->getTH2F() )->SetBins(24, 0, 24, nNumAMC, 0, nNumAMC);
+    ( (TH2F *)ChamT2D->getTH2F() )->GetYaxis()->SetBinLabel(nNumAMC,tmpLabel);
+  
+    ( (TH2F *)OOSG2D->getTH2F() )->SetBins(1, 0, 1, nNumAMC, 0, nNumAMC);
+    ( (TH2F *)OOSG2D->getTH2F() )->GetYaxis()->SetBinLabel(nNumAMC,tmpLabel);
+  }
+    
+  return mlAMCID[BID_];
+}
+
+int GEMDQMStatusDigi::GEBBinN(uint16_t BID_){
+  if(mlGEBID.find(BID_) == mlGEBID.end()){
+    int addIdx = mlGEBID.size();
+    mlGEBID[BID_] = addIdx;
+    
+    int nNumGEB = (int)mlGEBID.size();
+    string strLabel = "BID: "+to_string(BID_);
+    const char* tmpLabel =  strLabel.data();
+    ( (TH2F *)h2B1010All->getTH2F() )->SetBins(15, 0x0 , 0xf, nNumGEB, 0, nNumGEB);
+    ( (TH2F *)h2B1010All->getTH2F() )->GetYaxis()->SetBinLabel(nNumGEB,tmpLabel);
+
+    ( (TH2F *)h2B1100All->getTH2F() )->SetBins(15, 0x0 , 0xf, nNumGEB, 0, nNumGEB);
+    ( (TH2F *)h2B1100All->getTH2F() )->GetYaxis()->SetBinLabel(nNumGEB,tmpLabel);
+ 
+    ( (TH2F *)h2B1110All->getTH2F() )->SetBins(15, 0x0 , 0xf, nNumGEB, 0, nNumGEB);
+    ( (TH2F *)h2B1110All->getTH2F() )->GetYaxis()->SetBinLabel(nNumGEB,tmpLabel);
+ 
+    ( (TH2F *)h2FlagAll->getTH2F() )->SetBins(15, 0x0 , 0xf, nNumGEB, 0, nNumGEB);
+    ( (TH2F *)h2FlagAll->getTH2F() )->GetYaxis()->SetBinLabel(nNumGEB,tmpLabel);
+ 
+    ( (TH2F *)h2CRCAll->getTH2F() )->SetBins(0xffff, -32768, 32768, nNumGEB, 0, nNumGEB);
+    ( (TH2F *)h2CRCAll->getTH2F() )->GetYaxis()->SetBinLabel(nNumGEB,tmpLabel);
+ 
+    ( (TH2F *)h2InputID->getTH2F() )->SetBins(31,  0x0 , 0b11111, nNumGEB, 0, nNumGEB);
+    ( (TH2F *)h2InputID->getTH2F() )->GetYaxis()->SetBinLabel(nNumGEB,tmpLabel);
+ 
+    ( (TH2F *)h2Vwh->getTH2F() )->SetBins(4095,  0x0 , 0xfff, nNumGEB, 0, nNumGEB);
+    ( (TH2F *)h2Vwh->getTH2F() )->GetYaxis()->SetBinLabel(nNumGEB,tmpLabel);
+
+    ( (TH2F *)h2Vwt->getTH2F() )->SetBins(4095,  0x0 , 0xfff, nNumGEB, 0, nNumGEB);
+    ( (TH2F *)h2Vwt->getTH2F() )->GetYaxis()->SetBinLabel(nNumGEB,tmpLabel);
+ 
+    ( (TH2F *)h2GEBError->getTH2F() )->SetBins(5, 0, 5, nNumGEB, 0, nNumGEB);
+    ( (TH2F *)h2GEBError->getTH2F() )->GetYaxis()->SetBinLabel(nNumGEB,tmpLabel);
+
+    ( (TH2F *)h2GEBWarning->getTH2F() )->SetBins(10,  0, 10, nNumGEB, 0, nNumGEB);
+    ( (TH2F *)h2GEBWarning->getTH2F() )->GetYaxis()->SetBinLabel(nNumGEB,tmpLabel);
+   
+  }
+  return mlGEBID[BID_];
+}
+
+//----------------------------------------------------------------------------------------------------
+
+void GEMDQMStatusDigi::dqmBeginRun(edm::Run const &, edm::EventSetup const &)
```

Comment:
please remove empty methods, you don't have to define them.

Reply:
it was done.